cpp/include/cuspatial/experimental/iterator_factory.cuh

```diff
@@ -151,7 +236,7 @@ template <typename Iter>
 auto vec_2d_iterator_to_output_interleaved_iterator(Iter d_points_begin)
```

Comment:
Can this be an overload of `make_vec_2d_iterator`? For consistency, I think iterator factories should start with `make` and end in `iterator`.

Reply:
This has the same number of input argument to the existing make_vec2d_iterator though. How will the compiler know which overload the find?

Reply:
The existing `make_vec2d_iterator` takes two iterator arguments (x and y). This one takes a single iterator argument.

Reply:
Part of this PR is refactoring `interleaved_to_vec_2d_iterator` into `make_vec2d_iterator` (single argument). For this I think it's best to be called `make_output_vec_2d_iterator`.

Reply:
Yes, sorry for the confusion, I meant output_iterator.